src/SprykerEco/Zed/Payone/Business/Distributor/OrderPriceDistributor.php

```diff
@@ -0,0 +1,139 @@
+
+        $orderTransfer->setTotals($totals->setGrandTotal($payonePayment->getAmount()));
+
+        return $orderTransfer;
+    }
+
+    /**
+     * @param int $paymentAmount
+     * @param int $grandTotalAmount
+     *
+     * @return float
+     */
+    protected function calculatePriceRatio(int $paymentAmount, int $grandTotalAmount): float
+    {
+        return $paymentAmount / $grandTotalAmount;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\OrderTransfer $orderTransfer
+     *
+     * @return \Generated\Shared\Transfer\PaymentTransfer|null
+     */
+    protected function findPayonePayment(OrderTransfer $orderTransfer): ?PaymentTransfer
+    {
+        $paymentTransfers = $orderTransfer->getPayments();
+
+        foreach ($paymentTransfers as $paymentTransfer) {
+            if ($paymentTransfer->getPaymentProvider() === PayoneConfig::PROVIDER_NAME) {
+                return $paymentTransfer;
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * @param int $unitPrice
+     * @param float $priceRatio
+     * @param float $roundingError
+     *
+     * @return int
+     */
+    protected function calculateRoundedPrice(int $unitPrice, float $priceRatio, float &$roundingError): int
+    {
+        $priceBeforeRound = ($unitPrice * $priceRatio) + $roundingError;
+        $priceRounded = (int)round($priceBeforeRound);
+        $roundingError = $priceBeforeRound - $priceRounded;
+
+        return $priceRounded;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\OrderTransfer $orderTransfer
+     * @param float $priceRatio
+     * @param float $roundingError
+     *
+     * @return void
+     */
+    protected function distributeOrderItemsPrices(
+        OrderTransfer $orderTransfer,
+        float $priceRatio,
+        float &$roundingError
+    ): void {
+        foreach ($orderTransfer->getItems() as $itemTransfer) {
+            $priceRounded = $this->calculateRoundedPrice(
+                $itemTransfer->getSumPriceToPayAggregation(),
+                $priceRatio,
+                $roundingError,
+            );
+
+            $itemTransfer->setSumPriceToPayAggregation($priceRounded);
+
+            $unitPrice = $priceRounded / $itemTransfer->getQuantity();
+            $itemTransfer->setUnitPriceToPayAggregation($unitPrice);
+            $itemTransfer->setUnitGrossPrice($unitPrice);
+        }
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\OrderTransfer $orderTransfer
+     * @param float $priceRatio
+     * @param float $roundingError
+     *
+     * @return void
+     */
+    protected function distributeOrderExpensesPrices(
+        OrderTransfer $orderTransfer,
+        float $priceRatio,
+        float &$roundingError
+    ): void {
+        foreach ($orderTransfer->getExpenses() as $expenseTransfer) {
+            $expenseTransfer->setSumGrossPrice(
+                $this->calculateRoundedPrice(
```

Comment:
Would be more readable if you save the result to a variable before setting.

Reply:
Fixed